Security/Core/Authentication/Provider/Provider.php

```diff
@@ -158,7 +162,7 @@ protected function getCurrentTime()
 
     protected function isTokenFromFuture($created)
     {
-        return strtotime($created) > strtotime($this->getCurrentTime());
+        return strtotime($created) - $this->clockSkew > strtotime($this->getCurrentTime());

```

Comment:
Well, i initially implemented something like this at first, but the opposite. Your solution would limit the lifetime with correctly synced clocks. I considered the clock skew to be a margin of acceptable error, the allowed time difference between the client clock and the systems clock. So the clock skew would have to be added to the current time, so that a client clock which would be in de past, would have an extended lifetime of clock skew seconds.

Then i thought, is this really a nice to have feature? For a client clock being set to 29 secs in to the future and a clock skew setting of 30 secs, this would mean extending the lifetime by 59 seconds, which didn't feel like something that would be desirable.

I came to the conclusion it would be better to determine the clock skew automatically by determining the difference between the "Created" timestamp of the requesting WSSE header and the servers current time. I was confused about if it would be possible to create something like this, and because the project i was working on was fixed by my patch, i moved on and forgot about it.

So for now i would suggest not using the clock skew for expiration time and maybe file a feature request for dynamic determination of the clock skew.